go/felix/iptables/table.go

```diff
@@ -630,18 +619,29 @@ func (t *Table) applyUpdates() error {
 				line := deleteRule(chainName, ruleNum)
 				inputBuf.WriteString(line)
 				inputBuf.WriteString("\n")
-			} else {
-				// Make sure currentHashes ends up the right length.
-				currentHashes = append(currentHashes, "")
 			}
 		}
-		for i := len(rules) - 1; i >= 0; i-- {
-			prefixFrag := t.commentFrag(currentHashes[i])
-			line := rules[i].RenderInsert(chainName, prefixFrag)
-			inputBuf.WriteString(line)
-			inputBuf.WriteString("\n")
+
```

Comment:
Code just above only removes, and the folllowing code uses reverse order for insert but normal order for append, so perhaps:
- update the comment above to say "remove" instead of "remove/insert"
- add a comment for the insert case below to explain that we use reverse order because (I guess) we always insert at position 1.